### PR TITLE
email validation for user input

### PR DESCRIPTION
**Pull request by tino097**
Fixes #
user input for email when creating datasets

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ x] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/logic/validators.py [hunk 1]
```diff
@@ -831,3 +831,10 @@ def empty_if_not_sysadmin(key, data, errors, context):
         return
 
     empty(key, data, errors, context)
+
+def email_validator(value, context):
+    '''Validate email input '''
+    if value:
+        if not re.match("\A(?P<name>[\w\-_]+)@(?P<domain>[\w\-_]+).(?P<toplevel>[\w]+)\Z",value,re.IGNORECASE):
```
Comment (TkTech) [inline ckan/logic/validators.py]: Line is too long, PEP8.

File: ckan/logic/validators.py [hunk 2]
```diff
@@ -831,3 +831,10 @@ def empty_if_not_sysadmin(key, data, errors, context):
...
+            raise Invalid(_('Email "%s" is not valid format' % (value)))
```
Comment (TkTech) [inline ckan/logic/validators.py]: This string can't be translated. Should be:

`_('Email "%s" is not valid format') % value`

But even better,

`_('Email {email} is not a valid format').format(email=email)`. It's good practice to use named blocks when doing translations.

File: ckan/logic/validators.py (hunk 1 above)
Comment (wardi) [inline ckan/logic/validators.py]: This is the one we're using, based on the HTM5 spec for email validation https://github.com/open-data/ckanext-canada/blob/master/ckanext/canada/validators.py#L112-L125

File: ckan/logic/validators.py (hunk 2 above)
Comment (TkTech) [inline ckan/logic/validators.py]: Actually, I just realized showing the failed validation string in a flash message is...not ideal...

Comment (tino097) [inline ckan/logic/validators.py]: what are you suggesting ?!

**Comment by amercader** [comment]
Tests complain `Email test_author@test_author.com is not a valid format` which seems pretty valid to me.

**Comment by adamamyl** [comment]
@amercader Underscores aren't valid in domain names (maybe use something from https://tools.ietf.org/html/rfc2606#section-3 ?).

**Comment by amercader** [comment]
Thanks @adamamyl! You learn something everyday :)

Then it seems like it's just a matter of updating the test.

**Comment by tino097** [comment]
@adamamyl thank you !!!

It's ok now

**Comment by amercader** [comment]
Thanks @tino097

**Pull request merged**